### PR TITLE
IKASAN-1511 - File transfer streams - to support large FTP/SFTP payloads without chunking

### PR DESCRIPTION
**Pull request by elliotkennedy**
I have been fairly brutal with deprecating methods on the payload object. Please let me know if you do not think a method should be deprecated.

### CODE REVIEW

File: ikasaneip/component/endpoint/filetransfer/ftp/src/main/java/org/ikasan/endpoint/ftp/FileTransferConnectionTemplate.java [hunk 1]
```diff
@@ -113,36 +113,6 @@ public FileTransferConnectionTemplate(ConnectionSpec connectionSpec,Transactiona
...
-     * @param payload - The payload to deliver
-     * @param outputDir - The directory to place the file in
-     * @param outputTargets - The Map of targets to deliver the file to
-     * @param overwrite - Overwrite existing files flag
-     * @param renameExtension - The extension for the temp file rename
-     * @param checksumDelivered - Flag for whether we perform checksumming
-     * @param unzip - Flag for whether we unzip the delivered file
-     * @param cleanup - Cleanup txn journal flag
-     * @throws ResourceException - Exception if JCA connector fails
-     * @deprecated - use deliverInputStream
```
Comment (elliotkennedy) [inline ikasaneip/component/endpoint/filetransfer/ftp/src/main/java/org/ikasan/endpoint/ftp/FileTransferConnectionTemplate.java]: These methods have been deprecated for a few years now.

**Comment by elliotkennedy** [comment]
> Elliot for my understanding, you removing deprecated methods, adding extra method to Payload interface for returning input stream. I could see as well extra implementation for Payload. Is the intention for the new Payload implementation to be used by all FTP SFTP producers/ templates by default?
> Does this change only impacts Producer side not consumer?

That's correct Andrzej, thanks for taking the time to look. 

I have not changed any classes to use the new payload, and I was not planning to, I think an in-memory byte[] implementation is probably a more useful payload in most cases - the file-backed payload is just available for those who want to use it and it is now compatible with the FTP/SfTP producers so they no longer have to implement custom producers themselves. 

I think that making the Payload interface InputStream-based makes sense, and will allow for different payload sources to be transmitted in the future but I would be keen to hear opinions on this, I have only gone with it because it felt like the best fit.

I have not made any changes to the consumer side, but only because I was not thinking about it. My initial thoughts are that it would be good to keep the memory footprint on the consumers low using this method, and only bring the contents into memory in downstream components when needed but I have not implemented anything on that front.

**Comment by andrzej-majewski** [comment]
@elliotkennedy 

I am happy with the changes. Just before we merge lets have a quick discussion with Jeff.

I will create Ikasan JIRA to cover above work.

**Comment by elliotkennedy** [comment]
Sweet thanks - happy to write it up if you can invite to confluence.

**Comment by mitcje** [comment]
@elliotkennedy  @andrzej-majewski  Lets have a quick chat so I am up to speed. It all seems sensible - whats the jira for this?

**Comment by andrzej-majewski** [comment]
@elliotkennedy  https://ikasan.atlassian.net/browse/IKASAN-1511 created to cover this.

Let have quick discussion on Tuesday and we get it merged then.

**Comment by elliotkennedy** [comment]
@andrzej-majewski @mitcje - Great thanks, let's talk Tuesday :+1:

**Pull request merged**